centreon/src/Core/ServiceTemplate/Domain/Model/ServiceTemplate.php

```diff
@@ -0,0 +1,541 @@
+    ) {
+        $className = (new \ReflectionClass($this))->getShortName();
+        Assertion::positiveInt($id, "{$className}::id");
+        foreach (
+            [
+                'name' => self::MAX_NAME_LENGTH,
+                'alias' => self::MAX_ALIAS_LENGTH,
+                'comment' => self::MAX_COMMENT_LENGTH,
+                'description' => self::MAX_DESCRIPTION_LENGTH,
+                'note' => self::MAX_NOTES_LENGTH,
+                'noteUrl' => self::MAX_NOTES_URL_LENGTH,
+                'actionUrl' => self::MAX_ACTION_URL_LENGTH,
+                'iconAlternativeText' => self::MAX_ICON_ALT_LENGTH,
+            ] as $field => $limitation
+        ) {
+            $propertyValue = ${$field};
+            if ($propertyValue !== null) {
+                $this->{$field} = trim($propertyValue);
+                Assertion::notEmptyString($this->{$field}, "{$className}::{$field}");
+                Assertion::maxLength($this->{$field}, $limitation, "{$className}::{$field}");
+            }
+        }
+
+        // Assertions on ForeignKeys
+        $foreignKeys = [
+            'serviceTemplateParentId',
+            'commandId',
+            'eventHandlerId',
+            'notificationTimePeriodId',
+            'checkTimePeriodId',
+            'iconId',
+            'graphTemplateId',
+            'severityId',
+        ];
+        foreach ($foreignKeys as $propertyName) {
+            $propertyValue = ${$propertyName};
+            if ($propertyValue !== null) {
+                Assertion::positiveInt($propertyValue, "{$className}::{$propertyName}");
+            }
+        }
+        $properties = [
+            'maxCheckAttempts',
+            'normalCheckInterval',
+            'retryCheckInterval',
+            'freshnessThreshold',
+            'notificationInterval',
+            'recoveryNotificationDelay',
+            'firstNotificationDelay',
+            'acknowledgementTimeout',
+            'lowFlapThreshold',
+            'highFlapThreshold',
+        ];
+        foreach ($properties as $propertyName) {
+            $propertyValue = ${$propertyName};
+            if ($propertyValue !== null) {
+                Assertion::min($propertyValue, 0, "{$className}::{$propertyName}");
+            }
+        }
+
+        $this->commandArguments = [];
+        foreach ($commandArguments as $argument) {
+            if (is_scalar($argument)) {
+                $this->commandArguments[] = (string) $argument;
+            }
+        }
+
+        $this->eventHandlerArguments = [];
+        foreach ($eventHandlerArguments as $argument) {
+            if (is_scalar($argument)) {
+                $this->eventHandlerArguments[] = (string) $argument;
+            }
+        }
+
+        foreach ($notificationTypes as $type) {
+            if (! ($type instanceof NotificationType)) {
+                Assertion::isInstanceOf($type, NotificationType::class, "{$className}::notificationTypes");
+            }
```

Comment:
Love that 😍 
One question : why doing the `instanceof` here when it is already done inside the `Assertion::isInstanceOf` ?
```suggestion
            Assertion::isInstanceOf($type, NotificationType::class, "{$className}::notificationTypes");
```